pkgs/development/tools/steamos-devkit/default.nix

```diff
@@ -0,0 +1,117 @@
+{ lib
+, fetchFromGitHub
+, fetchFromGitLab
+, writeScript
+, python3
+, makeDesktopItem
+, pkg-config
+, SDL2
+}:
+let
+  # steamos-devkit requires a build of the unreleased pyimgui 2.0 branch, move to pythonPackages when 2.0 is released.
+  pyimgui = python3.pkgs.buildPythonPackage rec {
+    name = "pyimgui";
+    version = "1f095af5886f424ee12f26fa93b108b6420fafa4"; # 2.0 dev
+
+    src = fetchFromGitHub {
+      owner = "pyimgui";
+      repo = "pyimgui";
+      rev = version;
+      fetchSubmodules = true;
+      sha256 = "sha256-k070ue132m8H1Zm8bo7J7spCS5dSTGOj689ci7vJ+aw=";
+    };
+
+    nativeBuildInputs = with python3.pkgs; [
+      cython
+      pkg-config
+      SDL2
+    ];
+
+    propagatedBuildInputs = with python3.pkgs; [
+      click
+      pyopengl
+      pysdl2
+    ];
+
+    buildPhase = ''
+      python setup.py build bdist_wheel
+    '';
+
+    # Requires OpenGL acceleration
+    doCheck = false;
+  };
+in
+python3.pkgs.buildPythonPackage rec {
+  pname = "steamos-devkit";
+  version = "v0.20221101";
+
+  src = fetchFromGitLab {
+    domain = "gitlab.steamos.cloud";
+    owner = "devkit";
+    repo = "steamos-devkit";
+    rev = version;
+    sha256 = "sha256-VKnfcMF3WxkvqxlhJF+5j4Hso/TZpSS4dMOmSrWagRU=";
+  };
+
+  propagatedBuildInputs = with python3.pkgs; [
+    appdirs
+    bcrypt
+    cffi
+    cryptography
+    idna
+    ifaddr
+    netifaces
+    numpy
+    paramiko
+    pycparser
+    pyimgui
+    pynacl
+    pysdl2
+    signalslot
+    six
+  ];
+
+  patchPhase = ''
+    cp setup/shiv-linux-setup.py setup.py
+    mv client/* .
+  '';
+
+  postInstall = ''
+    mkdir -p $out/bin
+
+    cp -R ./devkit-utils $out/lib/python3.10/site-packages/devkit-utils
+
+    cp ${writeScript "steamos-devkit" ''
+      #!${python3}/bin/python3
+      import os
+
+      # Workaround to get pysdl to work on wayland, remove when https://gitlab.steamos.cloud/devkit/steamos-devkit/-/issues/1 is solved.
+      if os.environ.get("XDG_SESSION_TYPE") == "wayland":
+        os.environ["SDL_VIDEODRIVER"] = "wayland"
+
+      import devkit_client.gui2
+      devkit_client.gui2.main()
+    ''} $out/bin/steamos-devkit
```

Comment:
This should be done in a let in and symlinked here

Reply:
Symlinking makes nix not patch the shebang to make the python packages accessible, but I have moved it to the outer let.

Reply:
but the shebang should be correct already.

Reply:
I guess patching shebang is the wrong term, I guess it will be more right to say that it won't wrap it which means the python packages are not available